### ISSUE TITLE
Is it possible to run suites in parallel using different threads ?

### ISSUE THREAD

**Issue by rhapsodyman** [robotframework/robotframework#2988]
I want to understand if it is possible to run suites using parallel threads. The question is - will robotframework internal code be thread safe in this case. I know that there is a pabot project - but it uses parallel processes.  

One more question - if we use a library with ROBOT_LIBRARY_SCOPE = 'GLOBAL'  - will the code below get a separate instance of library for each run() method or only one global instance will be created in current process ?

Code will be as follows:
```
from robot import run
from multiprocessing.pool import ThreadPool

# create pool ...
result = pool.map_async(execute_in_same_process(suites) # execute for each suite

...
def execute_in_same_process(args):
   # get arguments ...
   rc = run(*datasources, stdout=stdout, stderr=stderr, **options) # robot run starts here
```

**Comment by pekkaklarck** [comment]
No, Robot unfortunately isn't threadsafe. Hopefully one day it is, and we've been trying to remove global state to move into that direction. Still a lot to do, and with the current resources we won't be able to accomplish that goal in the foreseeable future.
